psi4/src/psi4/libmints/wavefunction.cc

```diff
@@ -753,6 +739,28 @@ void Wavefunction::initialize_singletons() {
     done = true;
 }
 
+const Dimension Wavefunction::doccpi(bool assume_socc_alpha) const {
+    std::vector<int> docc_vec;
+    for (int h = 0; h < nalphapi_.n(); h++) {
+        docc_vec.push_back(std::min(nalphapi_[h], nbetapi_[h]));
+        if (assume_socc_alpha && nbetapi_[h] > nalphapi_[h]) {
+            outfile->Printf("Warning! Irrep has more beta than alpha electrons.\n");
+        }
+    }
+    return docc_vec;
+}
+/// Returns the SOCC per irrep array. Not recommended for unrestricted code.
+const Dimension Wavefunction::soccpi(bool assume_socc_alpha) const {
+    std::vector<int> socc_vec;
+    for (int h = 0; h < nalphapi_.n(); h++) {
+        socc_vec.push_back(std::abs(nalphapi_[h] - nbetapi_[h]));
+        if (assume_socc_alpha && nbetapi_[h] > nalphapi_[h]) {
+            outfile->Printf("Warning! Irrep has more beta than alpha electrons.\n");
+        }
```

Comment:
`assume_socc_alpha` is not properly handled here

Reply:
How? I don't see an issue. If you assume that all SOCC are alpha _and_ observe that there are more beta than alpha for a given irrep, you have a problem. That is exactly the problem I am trying to catch.

Reply:
My issue is that it's not clear what the code should be doing for `assume_socc_alpha = false`.

Reply:
That we are not assuming all SOCC are alpha does not imply that we are assuming all SOCC are not alpha. If you can find a better argument name, I can change it. `warn_on_beta_socc`?

Reply:
@susilehtola

Reply:
`bool sanity_check`?

Reply:
The name `sanity_check` is only appropriate if having more beta than alpha is _always_ inappropriate when this function is called. Lori and I talked among ourselves and thought it was best not to make that assumption. If that assumption _were_ correct, we should raise an error rather than warn, anyways.

Reply:
Renaming to `warn_on_beta_socc`, given aforementioned objections to `sanity_check`. Happy to re-discuss if there's a new alternative.